internal/collect/datacenters.go

```diff
@@ -52,7 +54,14 @@ func Datacenters(config *load.Config) {
 func collectEvents(config *load.Config, d mo.Datacenter, newDatacenter *load.Datacenter) {
 	//https://pubs.vmware.com/vsphere-51/index.jsp?topic=%2Fcom.vmware.wssdk.apiref.doc%2Fvim.HistoryCollector.html
 
-	c := cache.NewCache(d.Name, config.CachePath)
+	path := persist.DefaultPath(config.IntegrationName)
+	store, err := persist.NewFileStore(path, nil, time.Hour*24)
+	// if for some reason we couldn't create a file store, use an in-memory store.
+	if err != nil {
+		config.Logrus.WithError(err).Warn("could not create file based cache for events. using in-memory store")
+		store = persist.NewInMemoryStore()
+	}
```

Comment:
wdyt to move this into the NewCache func ?

Reply:
I had it inside before, but that makes the tests harder, because it automatically created disk store and some folders that may make the tests fail because of permissions or something else. I prefer to pass the store dependency into the cache so we can choose if we want a InMemory store or a File Store. PErhaps we can move this code to another place instead of here in the "datacenters" module